external_search.py

```diff
@@ -2331,6 +2332,53 @@ def __init__(self, collections=None, media=None, languages=None,
         else:
             self.scoring_functions = []
 
+    @property
+    def audiences(self):
+        """Return the appropriate audiences for this query.
```

Comment:
Filter.audiences was previously a static property. We need a `TestFilter.test_audiences` that sets `filter._audiences` to a bunch of different values and then verifies that `filter.audiences` gives the right result.